addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -2176,4 +2181,32 @@ MockServer.include({
             starred_counter: this._getRecords('mail.message', [['starred_partner_ids', 'in', user.partner_id]]).length,
         };
     },
+    /**
+     * Simulate the `notify_cancel_by_type` on `mail.thread` .
+     * Note that this method is overridden by snailmail module but not simulated here.
+     */
+    _mockMailThreadNotifyCancelByType(model, notificationType) {
+        const self = this;
+        const authorId = this.currentPartnerId;
+
+        let notificationMails = this._getRecords('mail.notification', [
+            ['notification_type', '=', notificationType],
+            ['notification_status', 'in', ['bounce', 'exception']],
+        ]).map(notification => ({
+            notification: notification,
+            mail: self._getRecords('mail.message', [['id', '=', notification.mail_message_id]])[0]
+        })).filter(notifMail => notifMail.mail.model === model && notifMail.mail.author_id === authorId);
+
+        notificationMails.forEach(notifMail => notifMail.notification.notification_status = 'canceled');
+        const mailIds = notificationMails.map(notifMail => notifMail.mail.id);
+
+        // Following lines mock mail.message._notify_message_notification_update on the selected mail
```

Comment:
Would be better to create a separate mock method for it in case we need it to use it elsewhere too as we extend coverage.

Reply:
This is a special case with a fixed user and not a full implementation of that (complex) method in javascript. I have changed the message by just telling we send notification on the bus to update the client.